gordon/main.py

```diff
@@ -83,20 +86,76 @@ def setup(config_root=''):
     return config
 
 
-def _log_or_exit_on_exceptions(errors, debug):
+def _log_or_exit_on_exceptions(base_msg, exc, debug):
     log_level_func = logging.warn
     if not debug:
         log_level_func = logging.error
 
-    base_msg = 'Plugin "{name}" was not loaded:'
-    for name, exc in errors:
-        msg = base_msg.format(name=name)
-        log_level_func(msg, exc_info=exc)
+    if isinstance(exc, list):
+        for exception in exc:
+            log_level_func(base_msg, exc_info=exception)
+    else:
+        log_level_func(base_msg, exc_info=exc)
 
     if not debug:
         raise SystemExit(1)
 
 
+def _assert_required_plugins(plugins, debug):
+    required_providers = {
+        'event_consumer': False,
+        'enricher': False,
+        'publisher': False,
+    }
+    for plugin in plugins:
+        if interfaces.IEventConsumerClient.providedBy(plugin):
+            required_providers['event_consumer'] = True
+        elif interfaces.IEnricherClient.providedBy(plugin):
+            required_providers['enricher'] = True
+        elif interfaces.IPublisherClient.providedBy(plugin):
+            required_providers['publisher'] = True
+
+    missing = []
+    msg = ('The provider for "{name}" interface is not configured for the '
+           'Gordon service or is not implemented.')
+    for provider, available in required_providers.items():
+        if not available:
+            exc = exceptions.MissingPluginError(msg.format(name=provider))
+            missing.append(exc)
+    if missing:
+        base_msg = 'Issue running plugins: '
+        _log_or_exit_on_exceptions(base_msg, missing, debug=debug)
+
+
+def _gather_runnable_plugins(plugins, debug):
+    plugins_to_run = []
+    for plugin in plugins:
+        if interfaces.IEventConsumerClient.providedBy(plugin):
+            # TODO (lynn): this should be switched out for adding
```

Comment:
I totally agree I think it's time to introduce it, but I feel like it's out of scope for the ticket I'm working on, unless you feel strongly otherwise.

Reply:
To expand: adding verification requires work to update the GCP plugin as well. So yeah I think verification should be its own task separate from this.

Reply:
^ Gotcha, makes sense.